app/models/reports/repository.rb

```diff
@@ -238,18 +239,23 @@ def denominator(region, period)
     #   }
     #
     def region_period_cached_query(calculation, &block)
+      results = regions.each_with_object({}) { |region, hsh| hsh[region.slug] = Hash.new(0) }
```

Comment:
Ensure we always have an entry for every region, even if there is no data for that particular region.